crates/astria-conductor/src/executor/mod.rs

```diff
@@ -437,8 +513,60 @@ impl Executor {
         Ok(executed_block)
     }
 
+    #[instrument(
+        skip_all,
+        fields(
+            firm_number = self.state.firm_number(),
+            soft_number = self.state.soft_number(),
+        )
+    )]
+    async fn populate_blocks_pending_finalization(
+        &mut self,
+        mut client: Client,
+    ) -> eyre::Result<()> {
+        if self.firm_blocks.is_none() || self.soft_blocks.is_none() {
```

Comment:
should this be an error case? Why would we call this before setting soft and firm block?

Reply:
`firm_blocks` and `soft_blocks` are the channels that are created during executor construction.

We use these as proxies for which "mode" executor operates under.